### ISSUE TITLE
DataPlane stagnant when ports are changed

### ISSUE THREAD

**Issue by shaneutt** [Kong/gateway-operator#118]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Currently if you create a `Gateway` resource, a `ControlPlane` and `DataPlane` will be created which have a `NetworkPolicy` restricting traffic to the `DataPlane`'s administrative APIs such that only the `ControlPlane` can communicate.

If the `Gateway` is updated to change the ports of the `DataPlane` communication can break down because the NetworkPolicy does not get updated.

### Expected Behavior

When ports are changed for the dataplane via `GatewayConfiguration` (or otherwise), connectivity from the `ControlPlane` to the `DataPlane` should be repaired.

**Comment by jrsmroz** [comment]
Dataplane ports are hardcoded. Not configurable.

**Comment by shaneutt** [comment]
If I'm not mistaken, we're heading in a direction where someone could set the `KONG_PROXY_LISTEN` or `KONG_ADMIN_LISTEN` manually if they so choose, that being the case the scope of this would include recognizing those changes, comprehending the values, and reflecting any change in port to the relevant network policies.

**Comment by jrsmroz** [comment]
If someone sets `KONG_PROXY_LISTEN` and `KONG_ADMIN_LISTEN` he still needs to change port mappings in the pods.

**Comment by shaneutt** [comment]
> If someone sets `KONG_PROXY_LISTEN` and `KONG_ADMIN_LISTEN` he still needs to change port mappings in the pods. Those mapping of exposed ports is hardcoded as of now. Just saying. If we support setting those env vars, we will needs to redo mechanism for creating deployments, so the ports are not hardcoded.

Indeed. I think a follow up for `v0.2.0` to decide on the exact behavior we want for port changing, AND to implement as needed would be a good way to move forward. This issue should be considered blocked on that issue.
